### PR TITLE
[VersionControl] Matches scrollarea in authors tab with its source editor

### PR DESCRIPTION
**Pull request by netonjm**
This issue, allow switch between Source code and Authors tab and respect the current line selected with the same scroll position

Fixes VSTS #596859 - usability: switching to Authors tab should jump to the context line, not to "cached" line state
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/596859

authors-fix

### CODE REVIEW

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs [hunk 1]
```diff
@@ -789,7 +786,33 @@ protected override bool OnExposeEvent (Gdk.EventExpose e)
...
+			internal void SetHighlight (Annotation annotation, int line, double y, bool needsRedraw)
+			{
+				highlightPositon = y;
+				highlightAnnotation = annotation;
+				if (needsRedraw) {
+					widget.QueueDraw ();
+				}
+			}
+
+			internal void Hightlight (int line)
```
Comment (Therzok) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs]: Typoe: Highlight

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs [hunk 2]
```diff
@@ -789,7 +786,33 @@ protected override bool OnExposeEvent (Gdk.EventExpose e)
 				}
 				return true;
 			}
-			
+
+			int GetStartLineFromY (double y) => widget.Editor.YToLine (widget.Editor.VAdjustment.Value + y);
```
Comment (Therzok) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs]: `YToStartLine`

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs (hunk 1 above)
Comment (netonjm) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs]: ops! :-)

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs [hunk 3]
```diff
@@ -789,7 +786,33 @@ protected override bool OnExposeEvent (Gdk.EventExpose e)
...
+
+			void GetAnnotationFromY (double y, out Annotation annotation, out int startLine)
+			{
+				startLine = GetStartLineFromY (y);
+				annotation = GetAnnotationFromLine (startLine);
+			}
+
+			internal Annotation GetAnnotationFromLine (int startLine) =>
+			 startLine > 0 && startLine <= annotations.Count ? annotations [startLine - 1] : null;
```
Comment (Therzok) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs]: indent?

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs [hunk 4]
```diff
@@ -789,7 +786,33 @@ protected override bool OnExposeEvent (Gdk.EventExpose e)
...
+			{
+				startLine = YToStartLine (y);
+				annotation = GetAnnotationFromLine (startLine);
+			}
+
+			internal Annotation GetAnnotationFromLine (int startLine) =>
+				startLine > 0 && startLine <= annotations.Count ? annotations [startLine - 1] : null;
+
+			internal void SetHighligh (Annotation annotation, int line, double y, bool needsRedraw)
+			{
```
Comment (rodrmoya) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs]: still a typo, it’s Highlight

**Comment by mkrueger** [comment]
Added code for syncing the folding & moved the adjustment set code. Adjustments can only be set after the text is set, otherwise the bounds are different.

**Comment by netonjm** [comment]
Great thanks @mkrueger !

**Pull request merged**